### PR TITLE
[SPARK-16455] Add a new hook in CoarseGrainedSchedulerBackend in order to stop scheduling new tasks when cluster is restarting

### PR DESCRIPTION
**Pull request by lovexi**
## What changes were proposed in this pull request?

In our case, we are implementing a new mechanism which will let driver survive when cluster is temporarily down and restarting. So when the service provided by cluster is not available, scheduler should stop scheduling new tasks. I added a hook inside CoarseGrainedSchedulerBackend class, in order to avoid new task scheduling when it's necessary.

JIRA link: https://issues.apache.org/jira/browse/SPARK-16455

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala [hunk 1]
```diff
@@ -345,6 +351,8 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
     driverEndpoint = createDriverEndpointRef(properties)
   }
 
+  def isClusterAvailableForNewOffers(): Boolean = true
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: document what this does?

Comment (lovexi) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala]: Done. Added doc for new hook to explain what it does.

**Comment by lovexi** [comment]
Ping

**Comment by rxin** [comment]
cc @kayousterhout

**Comment by kayousterhout** [comment]
@rxin I'm in favor of not merging this and commented on the JIRA (comment there if there's a reason I missed that this makes sense to add?)

**Pull request closed without merging**